lib/workers/repository/update/pr/code-owners.ts

```diff
@@ -5,9 +5,100 @@ import { readLocalFile } from '../../../../util/fs';
 import { getBranchFiles } from '../../../../util/git';
 import { newlineRegex, regEx } from '../../../../util/regex';
 
+interface FileOwnerRule {
+  usernames: string[];
+  pattern: string;
+  score: number;
+  match: (path: string) => boolean;
+}
+
+function extractOwnersFromLine(line: string): FileOwnerRule {
+  const [pattern, ...usernames] = line.split(regEx(/\s+/));
+  const matchPattern = ignore().add(pattern);
+  return {
+    usernames,
+    pattern,
+    score: pattern.length,
+    match: (path: string) => matchPattern.ignores(path),
+  };
+}
+
+interface UserScore {
+  username: string;
+  score: number;
+}
+
+interface FileScore {
+  file: string;
+  score: number;
+}
+
+interface FileOwnersScore {
+  file: string;
+  usernames: UserScore[];
+}
+
+function matchFileToOwners(
+  file: string,
+  rules: FileOwnerRule[]
+): FileOwnersScore {
+  const usersWithScore = rules
```

Comment:
please use for-of loop

Reply:
instead of using `reduce`? is there any reason for that change? imo it's cleaner with reduce

Reply:
code is cleaner with for-of loop, also reduces nesting.

Reply:
Solved in d2b3880f735ac5fd869d1013aab5207209eeea28